server/src/main/java/org/eclipse/hono/application/HonoApplication.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Copyright (c) 2016 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.application;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import org.eclipse.hono.authentication.AuthenticationService;
+import org.eclipse.hono.authorization.AuthorizationService;
+import org.eclipse.hono.config.HonoConfigProperties;
+import org.eclipse.hono.registration.RegistrationService;
+import org.eclipse.hono.server.HonoServer;
+import org.eclipse.hono.server.HonoServerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * The Hono server main application class.
+ * <p>
+ * This class configures and wires up Hono's components (Verticles).
+ * By default there will be as many instances of each verticle created as there are CPU cores
+ * available. The {@code hono.maxinstances} config property can be used to set the maximum number
+ * of instances to create. This may be useful for executing tests etc.
+ * </p>
+ */
+@Component
```

Comment:
I think we should remove this annotation so that this class gets not instantiated automatically. Users should add this annotation to their subclass instead. This way users have a choice to either create a subclass or create their own, independent application class.